src/coreclr/vm/methodtable.h

```diff
@@ -393,10 +417,30 @@ struct MethodTableWriteableData
         m_dwFlags &= ~(MethodTableWriteableData::enum_flag_HasApproxParent);
 
     }
-};  // struct MethodTableWriteableData
 
-typedef DPTR(MethodTableWriteableData) PTR_MethodTableWriteableData;
-typedef DPTR(MethodTableWriteableData const) PTR_Const_MethodTableWriteableData;
+    // The NonVirtualSlots array grows backwards, so this pointer points at just AFTER the first entry in the array
+    // To access, use a construct like... GetNonVirtualSlotsArray(pWriteableData)[-(1 + index)]
+    static inline PTR_PCODE GetNonVirtualSlotsArray(PTR_Const_MethodTableWriteableData pWriteableData)
+    {
+        LIMITED_METHOD_DAC_CONTRACT;
+        return dac_cast<PTR_PCODE>(dac_cast<TADDR>(pWriteableData) + pWriteableData->GetOffsetToNonVirtualSlots());
+    }
+
+    inline int16_t GetOffsetToNonVirtualSlots() const
+    {
+        return m_offsetToNonVirtualSlots;
+    }
+
+    inline void SetOffsetToNonVirtualSlots(int16_t offset)
+    {
+        m_offsetToNonVirtualSlots = offset;
+    }
+
+    static inline PTR_GenericsStaticsInfo GetGenericStaticsInfo(PTR_Const_MethodTableWriteableData pWriteableData)
+    {
+        return dac_cast<PTR_GenericsStaticsInfo>(dac_cast<TADDR>(pWriteableData) - sizeof(GenericsStaticsInfo));
+    }
+};  // struct MethodTableWriteableData
```

Comment:
Should `MethodTableWriteableData` be renamed to something that better captures its current purpose? Like `MethodTableAuxiliary` data?

Reply:
Ah, great, that's such a nice name. I didn't have a good one in mind, so I wasn't changing it, but MethodTableAuxiliaryData is a distinct improvement, especially since the writeable concept doesn't make any sense anymore.